src/pacman/sync.c

```diff
@@ -790,29 +791,30 @@ static int core_update(int *needed)
 		return 1;
 	}
 
-	*needed = 0;
+	needed = 0;
 	for(i = alpm_trans_get_add(config->handle); i; i = i->next) {
 		alpm_pkg_t *pkg = i->data;
 		if (alpm_pkg_is_core_package(pkg)) {
 			core = alpm_list_add(core, pkg);
-			*needed = 1;
+			needed = 1;
 		}
 	}
 
-	if(!(*needed)) {
+	if(!needed) {
 		if (!config->print) {
 			printf(_(" there is nothing to do\n"));
 		}
 		return 0;
 	}
 
 	/* hooks are most likely doomed to fail in a core upgrade, so disable them */
-	alpm_option_set_hookdirs(config->handle, NULL);
+	if (!(config->print || config->op_s_downloadonly || strcmp(config->rootdir, "/") != 0))
+		alpm_option_set_hookdirs(config->handle, NULL);
 
 	config->handle->trans->add = core;
 	pm_printf(ALPM_LOG_WARNING, _("terminate other MSYS2 programs before proceeding\n"));
 	retval = sync_prepare_execute();
-	if(retval == 0) {
+	if(retval == 0 && !(config->print || config->op_s_downloadonly || strcmp(config->rootdir, "/") != 0)) {
```

Comment:
nitpick: could we move "!(config->print || config->op_s_downloadonly || strcmp(config->rootdir, "/") != 0)" into a var, and also skip the warning in that case?